Metalama.Framework.DesignTime/Pipeline/AspectPipelineResult.cs

```diff
@@ -276,6 +277,21 @@ private bool IsEmpty
         var introducedTrees = introducedSyntaxTreeBuilder?.ToImmutable() ?? this.IntroducedSyntaxTrees;
         var inheritableAspects = inheritableAspectsBuilder?.ToImmutable() ?? this._inheritableAspects;
 
+        if ( externalValidators != null )
+        {
+            validatorsBuilder ??= this.ReferenceValidators.ToBuilder();
+
+            foreach ( var externalValidator in externalValidators )
+            {
+                if (externalValidator == null)
```

Comment:
Based on the nullability of the collection, this check shouldn't be necessary.

If it is necessary, I think you should either prevent `null`s from getting into a non-null collection, or make the collection type nullable: `IReadOnlyList<DesignTimeReferenceValidatorInstance?>?`.